backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt

```diff
@@ -486,11 +491,15 @@ internal class IntrinsicGenerator(private val environment: IntrinsicGeneratorEnv
 
         return when (val typeKind = LLVMGetTypeKind(first.type)) {
             llvm.LLVMTypeKind.LLVMFloatTypeKind, llvm.LLVMTypeKind.LLVMDoubleTypeKind -> {
-                val numBits = llvm.LLVMSizeOfTypeInBits(codegen.llvmTargetData, first.type).toInt()
-                val integerType = LLVMIntTypeInContext(llvmContext, numBits)!!
+                val integerType = LLVMIntTypeInContext(llvmContext, first.type.sizeInBits())!!
                 icmpEq(bitcast(integerType, first), bitcast(integerType, second))
             }
             llvm.LLVMTypeKind.LLVMIntegerTypeKind, llvm.LLVMTypeKind.LLVMPointerTypeKind -> icmpEq(first, second)
+            LLVMTypeKind.LLVMVectorTypeKind -> {
+                assert(first.type.sizeInBits() == 128)
+                val integerType = LLVMInt128TypeInContext(llvmContext)!!
```

Comment:
Isn't `LLVMInt128TypeInContext(llvmContext)` the same as `LLVMIntTypeInContext(llvmContext, 128)`?
If it is, why not just add `LLVMTypeKind.LLVMVectorTypeKind` to the first branch?